### PR TITLE
feat(curriculum): add searching and sorting algorithm based questions

### LINKED ISSUE #56215: [HACKTOBERFEST 2024] - Searching and Sorting Algorithms Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Searching and Sorting Algorithms quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-searching-and-sorting-algorithms`.

**All quiz questions need to relate to one of these topics:**

- What is an algorithm
- Runtime complexity and big-O notation
- binary search and linear search
- divide and conquer
- merge sort
- bubble sort
- quick sort

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by harshitaadam**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56215

### CODE REVIEW

**Comment by ilenia-magoni** [comment]
the quiz should have exactly 20 questions not 30

File: curriculum/challenges/english/25-front-end-development/quiz-searching-and-sorting-algorithms/66edd476078cf3f3bc654271.md [hunk 1]
```diff
@@ -17,439 +17,476 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder distractor 3
+O(1)
 
 #### --answer--
 
-Placeholder answer
+O(log n)
+
+---
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-searching-and-sorting-algorithms/66edd476078cf3f3bc654271.md:38]:
```suggestion
O(log n)
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Before I can review this in depth, you've added extra separators that should not be present at the end of each question set.

I've shown an example in the review suggestion.

**Comment by harshitaadam** [comment]
Ok I will resolve this

**Review by naomi-lgbt — APPROVED** [review]

**Review by gikf — APPROVED** [review]

**Pull request merged**